Emrald-UI/src/contexts/StateContext.tsx

```diff
@@ -57,23 +66,38 @@ export function useStateContext() {
   return context;
 }
 
-const StateContextProvider: React.FC<EmraldContextWrapperProps> = ({ children }) => {
-  const [states, setStates] = useState<State[]>(JSON.parse(JSON.stringify(appData.value.StateList.sort((a,b) => a.name.localeCompare(b.name)))));
+const StateContextProvider: React.FC<EmraldContextWrapperProps> = ({
+  children,
+}) => {
+  const [states, setStates] = useState<State[]>(
+    JSON.parse(
+      JSON.stringify(
+        appData.value.StateList.sort((a, b) => a.name.localeCompare(b.name)),
+      ),
+    ),
+  );
   const statesList = useComputed(() => appData.value.StateList);
 
   // Create, Delete, Update individual States
   const createState = (newState: State) => {
-    const updatedStates = [...states, newState];
+    const updatedStates = [...appData.peek().StateList, newState];
```

Comment:
This appData.peek() is something that I was testing out at one point but I'm not sure if it does what we want. Might be something good to research more about.